src/hypertable.c

```diff
@@ -308,3 +362,64 @@ hypertable_validate_triggers(PG_FUNCTION_ARGS)
 
 	PG_RETURN_VOID();
 }
+
+TS_FUNCTION_INFO_V1(hypertable_check_associated_schema_permissions);
+
+/*
+ * Check that the current user can create chunks in a hypertable's associated
+ * schema. This functions is typically called from create_hypertable() to
+ * verify that the user has the proper permissions.
+ */
+Datum
+hypertable_check_associated_schema_permissions(PG_FUNCTION_ARGS)
+{
+	Name		schema_name;
+	Oid			schema_oid;
+	Oid			user_oid;
+
+	if (PG_NARGS() != 2)
+		elog(ERROR, "Invalid number of arguments");
+
+	/*
+	 * If the schema name is NULL, it implies the internal catalog schema and
+	 * anyone can create chunks there
+	 */
+	if (PG_ARGISNULL(0))
+		PG_RETURN_NULL();
```

Comment:
May be my PG internals ignorance, but why is this PG_RETURN_NULL instead of PG_RETURN_VOID like the one further down where it checks if it's the internal schema?

Reply:
No ignorance involved. It should return void, as further down.